README.md

```diff
@@ -30,153 +31,238 @@ addOne(2); // 3
+acceptBA(printAB~(?1, ?0));             // prints: a, b
 ```
 
-is roughly identical in its behavior to:
+## Fixed Arity
+
+By default, partial application uses a fixed argument list: Normal arguments are evaluated and bound
+to their respective argument position, and placeholder (`?`) arguments and Ordinal placeholder (`?0`, etc.)
+arguments are bound to specific argument positions in the resulting partially applied function. As a result,
+excess arguments passed to a partially applied function have no specific position in which they should be 
+inserted. While this behavior differs from `f.bind()`, a fixed argument list allows us to avoid unintentionally
+accepting excess arguments:
 
 ```js
-const g = (() => {
-  const fn = f;
-  const p0 = 1;
-  return (a0, a1) => fn(a0, p0, a1);
-})();
+// (a)
+[1, 2, 3].forEach(console.log.bind(console, "element:"));
+// prints:
+// element: 1 0 1,2,3
+// element: 2 1 1,2,3
+// element: 3 2 1,2,3
+
+// (b)
+[1, 2, 3].forEach(x => console.log("element:", x));
+// prints:
+// element: 1
+// element: 2
+// element: 3
+
+// (c)
+[1, 2, 3].forEach(console.log~("element:", ?));
+// prints:
+// element: 1
+// element: 2
+// element: 3
+```
+
+In the example above, (a) prints extraneous information due to the fact that `forEach` not only passes the
+value of each element as an argument, but also the ordinal index of the element and the array in which the 
+element is contained.
+
+In the case of (b), the arrow function has a fixed arity. No matter how many excess arguments are passed to
+the callback, only the `x` parameter is forwarded onto the call.
+
+The intention of partial application is to emulate a normal call like `console.log("element:", 1)`, where 
+evaluation of the "applied" portions occurs eagerly with only the placeholders being "unapplied". This means 
+that excess arguments have no place to go as part of evaluation. As a result, (c) behaves similar to (b) in 
+that only a single argument is accepted by the partial function application and passed through to `console.log`.
+
+## Variable Arity: Pass Through Remaining Arguments using `...` 
+
+However, sometimes you may need the variable arity of something like `f.bind()`. To support this, partial
+application includes a `...` placeholder token with a specific meaning: Take the _rest_ of the arguments
+supplied to the partial function and _spread_ them into this position:
+
+```js
+const writeLog = (header, ...args) => console.log(header, ...args);
+const writeAppLog = writeLog~("[app]", ...);
+writeAppLog("Hello", "World!");             // prints: [app] Hello World!
+
+const writeAppLogWithBreak = writeAppLog~(..., "\n---\n");
+writeAppLogWithBreak("End of section");     // prints: [app] End of section\n---\n
 ```
 
-As well, with template expressions:
+A partially applied call may only have a single `...` in its argument list, though it may spread in other arguments
+using `...expr` as you might in a normal call.
```

Comment:
Can you give an example of this? I think this is saying that `foo~(?, ...arr, ?)` is supported?

Reply:
Yes. The `...arr` is part of a normal invocation, so it would be allowed. Essentially, the value `arr` would be spread into a List of applied parameters to `foo` when `foo~(?, ...arr, ?)` is evaluated so that the iterator isn't reevaluated for each invocation of the resulting partially applied function.

I will add an example.